src/main/java/com/onegini/examples/resourcegateway/web/ResourcesController.java

```diff
@@ -1,80 +1,98 @@
+  @GetMapping(value = "/devices")
+  public ResponseEntity<DeviceList> getDevices(@RequestHeader(AUTHORIZATION) final String authorizationHeader) {
+    final TokenIntrospectionResult tokenIntrospectionResult = getTokenIntrospectionResultFromHeader(authorizationHeader);
+    validateScopeAndTokenType(tokenIntrospectionResult, SCOPE_READ);
 
     return deviceApiRequestService.getDevices(tokenIntrospectionResult.getSub());
   }
 
-  @RequestMapping(value = "/application-details", method = RequestMethod.GET)
-  public ResponseEntity<?> getApplicationDetails(@RequestHeader(AUTHORIZATION) final String authorizationHeader) {
-
-    final String accessToken = accessTokenExtractor.extractFromHeader(authorizationHeader);
-    final TokenIntrospectionResult tokenIntrospectionResult = tokenIntrospectionService.introspectAccessToken(accessToken);
-
-    scopeValidationService.validateApplicationDetailsScopeGranted(tokenIntrospectionResult.getScope());
-    tokenTypeValidationService.validateNoImplicitAuthenticationToken(tokenIntrospectionResult.getAmr());
+  @GetMapping(value = "/application-details")
+  public ResponseEntity<ApplicationDetails> getApplicationDetails(@RequestHeader(AUTHORIZATION) final String authorizationHeader) {
+    final TokenIntrospectionResult tokenIntrospectionResult = getTokenIntrospectionResultFromHeader(authorizationHeader);
+    validateScopeAndTokenType(tokenIntrospectionResult, SCOPE_APPLICATION_DETAILS);
 
     final ApplicationDetails applicationDetails = new ApplicationDetails(tokenIntrospectionResult.getAppIdentifier(), tokenIntrospectionResult.getAppPlatform(),
         tokenIntrospectionResult.getAppVersion());
 
-    return new ResponseEntity<Object>(applicationDetails, OK);
+    return new ResponseEntity<>(applicationDetails, OK);
   }
 
-  @RequestMapping(value = "/user-id-decorated", method = RequestMethod.GET)
-  public ResponseEntity<?> getDecoratedUserId(@RequestHeader(AUTHORIZATION) final String authorizationHeader) {
-
-    final String accessToken = accessTokenExtractor.extractFromHeader(authorizationHeader);
-    final TokenIntrospectionResult tokenIntrospectionResult = tokenIntrospectionService.introspectAccessToken(accessToken);
+  @GetMapping(value = "/user-id-decorated")
+  public ResponseEntity<DecoratedUser> getDecoratedUserId(@RequestHeader(AUTHORIZATION) final String authorizationHeader) {
+    final TokenIntrospectionResult tokenIntrospectionResult = getTokenIntrospectionResultFromHeader(authorizationHeader);
 
     tokenTypeValidationService.validateImplicitAuthenticationToken(tokenIntrospectionResult.getAmr());
 
-    final DecoratedUserId decoratedUserId = new DecoratedUserIdBuilder()
+    final DecoratedUser decoratedUser = new DecoratedUserIdBuilder()
         .withUserId(tokenIntrospectionResult.getSub())
         .build();
 
-    return new ResponseEntity<>(decoratedUserId, OK);
+    return new ResponseEntity<>(decoratedUser, OK);
+  }
+
+  @PostMapping(value = "upload")
```

Comment:
Shouldn't we prefix the path with a slash here?

Reply:
Both work but I'll add it for consistency

Reply:
Yes, you are right 👍  I was also thinking about the consistency here 🙂